src/vivarium_census_prl_synth_pop/results_processing/addresses.py

```diff
@@ -166,19 +189,26 @@ def get_city_map(
     # Load addresses data from artifact
     addresses = artifact.load(data_keys.SYNTHETIC_DATA.ADDRESSES).reset_index()
     # Get observer data to map
-    output_cols = [column_name, "state"]
-    city_data = obs_data.reset_index()[output_cols].drop_duplicates().set_index("address_id")
+    if column_name == "address_id":
+        state_col = "state"
+    else:
+        state_col = "employer_state"
+    output_cols = [column_name, state_col]
+    city_data = obs_data.reset_index()[output_cols].drop_duplicates().set_index(column_name)
 
-    for state in city_data["state"].str.lower().unique():
+    for state in city_data[state_col].str.lower().unique():
         cities = vectorized_choice(
             options=addresses.loc[addresses["Province"] == state, "Municipality"],
-            n_to_choose=len(city_data.loc[city_data["state"] == state.upper()]),
+            n_to_choose=len(city_data.loc[city_data[state_col] == state.upper()]),
             randomness_stream=randomness,
-            additional_key=f"{additional_key}city_map",
+            additional_key=f"{additional_key}{column_name}_city_map",
         ).to_numpy()
-        city_data.loc[city_data["state"] == state.upper(), "city"] = cities
+        city_data.loc[city_data[state_col] == state.upper(), "city"] = cities
```

Comment:
This should be `employer_city` for businesses

Reply:
This is just creating a column and we are handling the column name when we add the key to the dictionary but I have changed it because I think it makes it a lot clearer.